### DISCUSSION TITLE
Check Boxes / Pivot

### DISCUSSION THREAD

**Discussion by 760524mkfa00** [livewire/livewire#1705] [4. General]
Hi,

I have the following where, Tags are pulled from a view composer to display all tags in a list on screen, on update I sync the checked ones with the user table using a pivot table:

`@foreach ($tags as $tag)
                <div class="py-1">
                    <div class="">
                        <input wire:model="newTags" type="checkbox" name="tags_id[]" value="{{ $tag->id }}" {{ ($this->originalTags->contains($tag->id)) ? 'checked="checked"' : '' }}>
                        {{ $tag->name }}
                    </div>
                </div>
            @endforeach`

In the Component I have the following:

`public $originalTags = [];
    public $newTags = [];

    public function mount(Student $student)
    {

        $this->student = $student;
        $this->originalTags = $this->student->tags()->get();
        
    }`

Without wire:model, it will check the related tags, however, as soon as bind it with wire:model no tags are checked.

Basically, it checks the correct checkboxes on load, but as soon as the binding happens it unchecks everything. I also tried:

`@foreach ($tags as $tag)
                <div class="py-1">
                    <div class="">
                        <input wire:model="originalTags" type="checkbox" name="tags_id[]" value="{{ $tag->id }}" >
                        {{ $tag->name }}
                    </div>
                </div>
            @endforeach`

To see if it would check the checkboxes, but nothing.

Would I be better querying the Tags table from within the mount function rather than a view composer and setting it on load? not sure which would be a good approach to this?

Thanks

**Comment by 760524mkfa00** [comment]
Hi, I got this worked out, the problem is when you wire to the check box you need to set the array values to true for the boxes you want checked, which makes absolute sense. 

This is what I ended up with:

Loop through all tags on the form
`@foreach ($tags as $tag)
                <div class="py-1">
                        <input wire:model="newTags.{{ $tag->id }}" type="checkbox">
                        {{ $tag->name }}
                </div>
            @endforeach`

In the Livewire component set the public property:
`public $newTags = [];`

Then in the Mount Method:

Get the checked boxes you want to be checked
`$this->originalTags = $this->student->tags()->get();`

set the checked boxes you want checked to true
`foreach($this->originalTags as $setTag)
{
$this->newTags[$setTag['id']] = true;
}`


Thanks
